store/src/main/java/com/nytimes/android/external/store/base/impl/BarcodeStoreBuilder.java

```diff
@@ -0,0 +1,110 @@
+package com.nytimes.android.external.store.base.impl;
+
+
+import com.nytimes.android.external.cache.Cache;
+import com.nytimes.android.external.store.base.DiskRead;
+import com.nytimes.android.external.store.base.DiskWrite;
+import com.nytimes.android.external.store.base.Fetcher;
+import com.nytimes.android.external.store.base.Parser;
+import com.nytimes.android.external.store.base.Persister;
+import com.nytimes.android.external.store.base.experimental.Store;
+import com.nytimes.android.external.store.util.NoopPersister;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import rx.Observable;
+import rx.functions.Func1;
+
+/**
+ * Builder where there parser is used.
+ */
+public class BarcodeStoreBuilder<Raw, Parsed, Key> {
+    private final List<Parser> parsers = new ArrayList<>();
+    private Persister<Raw, Key> persister;
+    private Cache<Key, Observable<Parsed>> memCache;
+    private Fetcher<Raw, Key> fetcher;
+
+    @NotNull
+    public static <Raw, Parsed, Key> BarcodeStoreBuilder<Raw, Parsed, Key> builder() {
+        return new BarcodeStoreBuilder<>();
+    }
+
+    @NotNull
+    public BarcodeStoreBuilder<Raw, Parsed, Key> fetcher(final @NotNull Fetcher<Raw, Key> fetcher) {
+        this.fetcher = fetcher;
+        return this;
+    }
+
+    @NotNull
+    public BarcodeStoreBuilder<Raw, Parsed, Key> persister(final @NotNull Persister<Raw, Key> persister) {
+        this.persister = persister;
+        return this;
+    }
+
+    @NotNull
+    public BarcodeStoreBuilder<Raw, Parsed, Key> persister(final @NotNull DiskRead<Raw, Key> diskRead,
+                                                           final @NotNull DiskWrite<Raw, Key> diskWrite) {
+        persister = new Persister<Raw, Key>() {
+            @NotNull
+            @Override
+            public Observable<Raw> read(Key barCode) {
+                return diskRead.read(barCode);
+            }
+
+            @NotNull
+            @Override
+            public Observable<Boolean> write(Key barCode, Raw raw) {
+                return diskWrite.write(barCode, raw);
+            }
+        };
+        return this;
+    }
+
+    @NotNull
+    public BarcodeStoreBuilder<Raw, Parsed, Key> parser(final @NotNull Func1<Raw, Parsed> parser) {
```

Comment:
Remove the `parser(final @NotNull Func1<Raw, Parsed> parser)` and keep `parser(final @NotNull Parser<Raw, Parsed> parser)`? For type safety :).